oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/measure/BanyanDBTopologyQueryDAO.java

```diff
@@ -303,25 +300,35 @@ List<Call.CallDetail> queryProcessRelation(Duration duration,
             timestampRange = new TimestampRange(TimeBucket.getTimestamp(startTB), TimeBucket.getTimestamp(endTB));
         }
         final String modelName = detectPoint == DetectPoint.SERVER ? ProcessRelationServerSideMetrics.INDEX_NAME :
-            ProcessRelationClientSideMetrics.INDEX_NAME;
-        final Map<String, Call.CallDetail> callMap = new HashMap<>();
+                ProcessRelationClientSideMetrics.INDEX_NAME;
+
         MeasureQueryResponse resp = query(modelName,
-                                          ImmutableSet.of(
-                                              Metrics.ENTITY_ID, ProcessRelationClientSideMetrics.COMPONENT_ID),
-                                          Collections.emptySet(), timestampRange, new QueryBuilder<MeasureQuery>() {
-                @Override
-                protected void apply(MeasureQuery query) {
-                    query.and(eq(ProcessRelationServerSideMetrics.SERVICE_INSTANCE_ID, serviceInstanceId));
+                ImmutableSet.of(Metrics.ENTITY_ID, ProcessRelationClientSideMetrics.COMPONENT_ID),
+                Collections.emptySet(), timestampRange, new QueryBuilder<MeasureQuery>() {
+                    @Override
+                    protected void apply(MeasureQuery query) {
+                        query.and(eq(ProcessRelationServerSideMetrics.SERVICE_INSTANCE_ID, serviceInstanceId));
+                        query.groupBy(Sets.newHashSet(Metrics.ENTITY_ID, ProcessRelationServerSideMetrics.COMPONENT_ID));
```

Comment:
* The HashSet doesn't maintain the order of tags. Could you opt for a sortable set instead? 
* Please add `component_id` to the measure's entity list. The order of tags in groupBy should match the order of entities. That will help improve the memory overhead

Reply:
You mean mark component_id and entity_id with `@SeriesID`?

Reply:
Yes. But the component_id doesn't exist in process_relation. From https://github.com/apache/skywalking/pull/10181#discussion_r1051883005, we could remove component_id from groupBy's tag list.

Reply:
> Yes. But the component_id doesn't exist in process_relation. From [#10181 (comment)](https://github.com/apache/skywalking/pull/10181#discussion_r1051883005), we could remove component_id from groupBy's tag list.

It is `ProcessRelationServerSideMetrics.COMPONENT_ID`. I think we need this in groupBy @wu-sheng, right?

Reply:
Oops, sorry.